sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -556,12 +556,14 @@ lsc_gather(AccessorTy acc, __ESIMD_NS::simd<uint32_t, N> offsets,
 /// given address, where S is a byte size of an "element" defined by the \c DS
 /// template parameter. The maximum size of accessed block is 512 bytes for PVC
 /// and 256 bytes for ACM (DG2).
-/// When \c DS equals \c lsc_data_size::u64, the address must be 8-byte aligned,
-/// otherwise - 4-bytes aligned. Allowed values for the data size are
-/// \c lsc_data_size::u32 and \c lsc_data_size::u64. Allowed NElts values are
-/// 1, 2, 3, 4, 8, 16, 32, 64.
-/// Note that to access 512 bytes, DS must be \c lsc_data_size::u64 and \c NElts
-/// must be 64.
+/// When \c DS equals \c lsc_data_size::u64 or \c sizeof(T) equal to 8 or DS is
+/// not set or set to \c lsc_data_size::default_size and the total amount of
+/// data i.e. \c sizeof(T) \c * \c NELTS is multiple of 8, the address must be
```

Comment:
Does this statement basically tells that lsc_block_load of simd<int, 2> with lsc_data_size::default_size, then the address 'p' must be aligned by 8? 
If so, then that seems wrong to me because for load of simd<int,2> still should be possible for address aligned by only 4.

Reply:
IMO, using bigger element size (assuming it requires also bigger address alignment) would require extension of this API by adding hint/guarantee from user that the address passed to this routine is properly aligned.
'overaligned_tag<8>' could be used for that purpose.